server/api/payments/paymentRouter.js

```diff
@@ -10,131 +10,109 @@ const router = require('express').Router();
+              customer: customer.id,
+              items: [{ plan }] // the value of plan is the id of the plan
+            },
+            (err, subscription) => {
+              // console.log('subscription \n', su`bscription);
+              err
+                ? res.send({ createdSubscription: false })
+                : res.send({
+                    createdCustomer: true,
+                    createdSubscription: true,
+                    stripeId: customer.id
+                  });
+            }
+          );
+  });
+});
 
 // Gets customer's plan using using stripeId
-router.get('/customer/plan', function(req, res) {
+router.post('/customer/plan', function(req, res) {
   // console.log("customer req \n", req.headers);
-  const { stripeid } = req.headers
-  if (!stripeid) {
-    res.send({
-      premium: false
-    })
-  }
-  stripe.customers.retrieve(
-    stripeid,
-    function(err, customer) {
+  const { stripeId } = req.body;
+
+  stripe.customers.retrieve(stripeId, function(error, customer) {
+    if (error) {
+      return res.status(500).send({ error });
+    }
+
+    if (customer) {
       if (customer.deleted) {
-        res.send({
-          message: "No subscription found",
+        return res.status(404).send({
+          message: 'No subscription found',
           premium: false
-        })
-        return;
-      } else if (err) {
-        res.send({ error: "Unable to get customer's plan"})
-      } else {
-        // console.log("customer plan \n", customer)
-        res.send({
-          customer: customer
-            .subscriptions
-            .data[0]
-            .items
-            .data[0]
-            .plan, 
-          premium: true  
-        })
+        });
       }
+      // console.log("customer plan \n", customer)
+      return res.status(200).send({
+        customer: customer.subscriptions.data[0].items.data[0].plan,
+        premium: true
+      });
     }
-  )
-})
+    return res.status(404).send({ error: 'Cannot find customer' });
+  });
+});
 
 // Gets customer's status; "premium: true" if customer has active subscription
 router.get('/customer/premium', function(req, res) {
   // console.log("customer req \n", req.headers);
-  const { stripeid } = req.headers
+  const { stripeid } = req.headers;
   if (!stripeid) {
     res.send({
       premium: false
-    })
+    });
   }
-  stripe.customers.retrieve(
-    stripeid, function(err, customer) {
-      if (customer.deleted) {
-        res.send({ premium: false })
-        return;
-      } else if (err) {
-        res.send({
-          error: "Unable to get customer"
-        })
-      } else {
-        // console.log("customer plan/n", customer)
-        res.send({
-          premium: customer
-            .subscriptions
-            .data[0]
-            .items
-            .data[0]
-            .plan
-            .active    
-        })
-      }
+  stripe.customers.retrieve(stripeid, function(err, customer) {
+    if (customer.deleted) {
```

Comment:
should we have a condition that asks if customer exists first like you did above?

Reply:
I missed this one. This route has not used anywhere yet, that is reason why I did not see the potential error.

But yea. It should.